Makefile

```diff
@@ -1,27 +1,96 @@
-FEATURES_INTERNALS=--features internals
+.DEFAULT_GOAL := help
+
+.PHONY: help
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+# -- variables --------------------------------------------------------------------------------------
+
+WARNINGS=RUSTDOCFLAGS="-D warnings"
+DEBUG_ASSERTIONS=RUSTFLAGS="-C debug-assertions"
 FEATURES_CONCURRENT_EXEC=--features concurrent,executable
 FEATURES_LOG_TREE=--features concurrent,executable,tracing-forest
 FEATURES_METAL_EXEC=--features concurrent,executable,metal
-PROFILE_OPTIMIZED=--profile optimized
-PROFILE_TEST=--profile test-release
 
-bench:
-	cargo bench $(PROFILE_OPTIMIZED)
+# -- linting --------------------------------------------------------------------------------------
+
+.PHONY: clippy
+clippy: ## Runs Clippy with configs
+	cargo +nightly clippy --workspace --all-targets --all-features -- -D warnings
+
+
+.PHONY: fix
+fix: ## Runs Fix with configs
+	cargo +nightly fix --allow-staged --allow-dirty --all-targets --all-features
+
+
+.PHONY: format
+format: ## Runs Format using nightly toolchain
+	cargo +nightly fmt --all
+
+
+.PHONY: format-check
+format-check: ## Runs Format using nightly toolchain but only in check mode
+	cargo +nightly fmt --all --check
+
+
+.PHONY: lint
+lint: format fix clippy ## Runs all linting tasks at once (Clippy, fixing, formatting)
+
+# --- testing -------------------------------------------------------------------------------------
+
+.PHONY: test
+test: ## Runs all tests
+	$(DEBUG_ASSERTIONS) cargo nextest run --cargo-profile test-release --features internals
+
+# --- docs ----------------------------------------------------------------------------------------
+
+.PHONY: doc
+doc: ## Generates & checks documentation
+	$(WARNINGS) cargo doc --all-features --keep-going --release
+
+.PHONY: mdbook
+mdbook: ## Generates mdbook
+	mdbook exec docs/
+
+# --- building ------------------------------------------------------------------------------------
+
+.PHONY: build
+build: ## Builds with default parameters
+	cargo build --release --features concurrent
+
+.PHONY: build-no-std
+build-no-std: ## Builds without the standard library
+	cargo build --no-default-features --target wasm32-unknown-unknown --workspace
+
+# --- executing ------------------------------------------------------------------------------------
+
+.PHONY: exec
+exec: ## Builds with optimized profile and features
```

Comment:
nit: I would change the comment to "Builds an executable with optimized profile and features". I would also make similar changes to other such commands.